### ISSUE TITLE
Runtime error when running with OMP and OpenCL

### ISSUE THREAD

**Issue by HariAbram** [AdaptiveCpp/AdaptiveCpp#1494]
When I run my SYCL code compiled with AdaptiveCPP(ACPP), I get a runtime error. ACPP compiler is only using the first available backend CPU driver in my case OpenCL, even when I pass `--acpp-targets=omp.accelerated` flag to compiler while compiling, it is compiling with openmp backend but it is not selecting openmp backend during runtime. 

```
=================Backend information===================
Loaded backend 0: OpenCL
  Found device: Intel(R) Xeon(R) CPU Max 9468
Loaded backend 1: OpenMP
  Found device: hipSYCL OpenMP host device
```

I get following error:
```
[AdaptiveCpp Error] from /home/AdaptiveCpp-24.02.0-ocl/include/hipSYCL/runtime/kernel_launcher.hpp:137 @ find_launcher(): No kernel launcher is present for requested backend
[AdaptiveCpp Error] from /home/AdaptiveCpp-24.02.0-ocl/src/runtime/ocl/ocl_queue.cpp:255 @ submit_kernel(): Could not obtain backend kernel launcher
```

**Comment by illuhad** [comment]
Right, I think this behavior is a remnant from before the times of the OpenCL backend. If there's no GPU, it just picks the first CPU - but since the introduction of OpenCL, that can happen to be an OpenCL device.

PS: Just use `--acpp-targets=generic` (which is also the default if there's no `--acpp-targets` argument). It's likely going to be faster than `omp.accelerated`, and can JIT code for all backends, including OpenMP.

**Comment by HariAbram** [comment]
Okay, thanks. I found a way around this by setting ACPP_VISIBILITY_MASK=omp, this will select OpenMP backend during runtime.

**Comment by illuhad** [comment]
Yes, the visibility mask is one way to affect which devices the default selector might pick.

I believe the original issue is fixed by #1501, please have a try :)
